src/demos/add-charts/app.js

```diff
@@ -2,87 +2,53 @@
 import React, {Component} from 'react';
 import MapGL from 'react-map-gl';
 import DeckGLOverlay from './deckgl-overlay';
-import LayerControls from './layer-controls';
+import {
+  LayerControls,
+  HEXAGON_CONTROLS
+} from './layer-controls';
 import Charts from './charts';
 import Spinner from './spinner';
 import {tooltipStyle} from './style';
 
-import taxiData from '../data/taxi.csv';
+import taxiData from '../data/taxi';
 
-const MAPBOX_STYLE = 'mapbox://styles/uberdata/cive485h000192imn6c6cc8fc';
+const MAPBOX_STYLE = 'mapbox://styles/mapbox/dark-v9';
 // Set your mapbox token here
 const MAPBOX_TOKEN = process.env.MapboxAccessToken; // eslint-disable-line
 
-const LAYER_CONTROLS = {
-  showHexagon: {
-    displayName: 'Show Hexagon',
-    type: 'boolean',
-    value: false
-  },
-  radius: {
-    displayName: 'Hexagon Radius',
-    type: 'range',
-    value: 250,
-    step: 50,
-    min: 50,
-    max: 1000
-  },
-  coverage: {
-    displayName: 'Hexagon Coverage',
-    type: 'range',
-    value: 0.7,
-    step: 0.1,
-    min: 0,
-    max: 1
-  },
-  upperPercentile: {
-    displayName: 'Hexagon Upper Percentile',
-    type: 'range',
-    value: 100,
-    step: 0.1,
-    min: 80,
-    max: 100
-  },
-  radiusScale: {
-    displayName: 'Scatterplot Radius',
-    type: 'range',
-    value: 30,
-    step: 10,
-    min: 10,
-    max: 200
-  }
-};
-
 export default class App extends Component {
 
   constructor(props) {
     super(props);
     this.state = {
       viewport: {
-        ...DeckGLOverlay.defaultViewport,
-        width: 500,
-        height: 500
+        width: window.innerWidth,
+        height: window.innerHeight,
+        longitude: -74,
+        latitude: 40.7,
+        zoom: 11,
+        maxZoom: 16
       },
-      points: [],
-      settings: Object.keys(LAYER_CONTROLS).reduce((accu, key) => ({
+      settings: Object.keys(HEXAGON_CONTROLS).reduce((accu, key) => ({
         ...accu,
-        [key]: LAYER_CONTROLS[key].value
+        [key]: HEXAGON_CONTROLS[key].value
       }), {}),
 
-      // hoverInfo
-      x: 0,
-      y: 0,
-      hoveredObject: null,
       status: 'LOADING'
     };
+    this._resize = this._resize.bind(this);
   }
 
   componentDidMount() {
     this._processData();
-    window.addEventListener('resize', this._resize.bind(this));
+    window.addEventListener('resize', this._resize);
     this._resize();
   }
 
+  componentWillUnmount() {
+    window.removeEventListener('resize', this._resize);
+  }
```

Comment:
if event listener is not removed, then there will be warnings each time people change section. 
just having addEventListener('resize', this._resize.bind(this)) then removeEventListener('resize', this._resize.bind(this)) is not enough because those 2 functions would not resolve to the same object. 
this._resize.bind(this) has to be stored in a variable. hence the line in the constructor.